File: quickbites-app/quickbites-backend/src/main/java/com/apsit/canteen_management/dto/UserResponseDto.java
package com.apsit.canteen_management.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserResponseDto {
    private String username;
    private String mobileNumber;
    private String email;
    private String profilePictureUrl;
}

File: quickbites-app/quickbites-user-app/canteen_user_app/lib/models/order_user/user_response_dto.dart
class UserResponseDto {
  final String username;
  final String mobileNumber;
  final String email;
  final String? profilePictureUrl;

  const UserResponseDto({
    required this.username,
    required this.mobileNumber,
    required this.email,
    this.profilePictureUrl,
  });

  factory UserResponseDto.fromJson(Map<String, dynamic> json) {
    return UserResponseDto(
      username: json['username'] as String,
      mobileNumber: json['mobileNumber'] as String,
      email: json['email'] as String,
      profilePictureUrl: json['profilePictureUrl'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'mobileNumber': mobileNumber,
      'email': email,
      'profilePictureUrl': profilePictureUrl,
    };
  }
}
